"Proposed API is only available..." error in v1.47

<!-- ⚠️⚠️ Do Not Delete This! bug_report_template ⚠️⚠️ -->
<!-- Please read our Rules of Conduct: https://opensource.microsoft.com/codeofconduct/ -->
<!-- Please search existing issues to avoid creating duplicates. -->
<!-- Also please test using the latest insiders build to make sure your issue has not already been fixed: https://code.visualstudio.com/insiders/ -->

<!-- Use Help > Report Issue to prefill these. -->
- VSCode Version: 1.47.0
- OS Version: macOS 10.15.4, Darwin x64 19.4.0

This may happen with more extensions, but I've only confirmed two.

With 1.47 installed, when I switch to the view container for Todo+ or TODO List the view breaks and nothing is shown. I receive the following error:

"[$extension]: Proposed API is only available when running out of dev or with the following command line switch: --enable-proposed-api $extension"

When I switch back to 1.46 I no longer receive the error and the extensions work as expected.

Steps to Reproduce:

1. Install 1.47
2. Install fabiospampinato.vscode-todo-plus or tzachovadia.todo-list
3. Restart VSCode
4. Switch to extension specific tree view

<!-- Launch with `code --disable-extensions` to check. -->
Does this issue occur when all extensions are disabled?: No

Thanks. I can now reproduce. Seeing the following error

```
/Users/jrieken/Code/vscode-all-stable-versions/Visual Studio Code-1.47.0.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2947 Error: [fabiospampinato.vscode-todo-plus]: Proposed API is only available when running out of dev or with the following command line switch: --enable-proposed-api fabiospampinato.vscode-todo-plus
	at a (/Users/jrieken/Code/vscode-all-stable-versions/Visual Studio Code-1.47.0.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:594)
	at Object.t.checkProposedApiEnabled (/Users/jrieken/Code/vscode-all-stable-versions/Visual Studio Code-1.47.0.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:594)
	at m.getTooltip (/Users/jrieken/Code/vscode-all-stable-versions/Visual Studio Code-1.47.0.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:611)
	at m.createTreeNode (/Users/jrieken/Code/vscode-all-stable-versions/Visual Studio Code-1.47.0.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:611)
	at m.createAndRegisterTreeNode (/Users/jrieken/Code/vscode-all-stable-versions/Visual Studio Code-1.47.0.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:610)
	at /Users/jrieken/Code/vscode-all-stable-versions/Visual Studio Code-1.47.0.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:609
	at processTicksAndRejections (/Users/jrieken/Code/vscode-all-stable-versions/Visual%20Studio%20Code-1.47.0.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/internal/process/task_queues.js:85)
	at async Promise.all (/Users/jrieken/Code/vscode-all-stable-versions/Visual%20Studio%20Code-1.47.0.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/index 0)
```

--- a/src/vs/workbench/api/common/extHostTreeViews.ts
+++ b/src/vs/workbench/api/common/extHostTreeViews.ts
@@ -24,6 +24,10 @@ import { IMarkdownString } from 'vs/base/common/htmlContent';
 
 type TreeItemHandle = string;
 
+function isMarkdownString(value: any): value is vscode.MarkdownString {
+	return (value !== undefined) && value.appendCodeblock && value.appendMarkdown && value.appendText && (value.value !== undefined);
+}
+
 function toTreeItemLabel(label: any, extension: IExtensionDescription): ITreeItemLabel | undefined {
 	if (isString(label)) {
 		return { label };
@@ -527,14 +531,11 @@ class ExtHostTreeView<T> extends Disposable {
 	}
 
 	private getTooltip(tooltip?: string | vscode.MarkdownString): string | IMarkdownString | undefined {
-		if (typeof tooltip === 'string') {
-			return tooltip;
-		} else if (tooltip === undefined) {
-			return undefined;
-		} else {
+		if (isMarkdownString(tooltip)) {
 			checkProposedApiEnabled(this.extension);
 			return MarkdownString.from(tooltip);
 		}
+		return tooltip;
 	}
 
 	private createTreeNode(element: T, extensionTreeItem: vscode.TreeItem2, parent: TreeNode | Root): TreeNode {

--- a/src/vs/workbench/api/common/extHostTreeViews.ts
+++ b/src/vs/workbench/api/common/extHostTreeViews.ts
@@ -13,7 +13,7 @@ import { ExtHostTreeViewsShape, MainThreadTreeViewsShape } from './extHost.proto
 import { ITreeItem, TreeViewItemHandleArg, ITreeItemLabel, IRevealOptions } from 'vs/workbench/common/views';
 import { ExtHostCommands, CommandsConverter } from 'vs/workbench/api/common/extHostCommands';
 import { asPromise } from 'vs/base/common/async';
-import { TreeItemCollapsibleState, ThemeIcon } from 'vs/workbench/api/common/extHostTypes';
+import { TreeItemCollapsibleState, ThemeIcon, MarkdownString as MarkdownStringType } from 'vs/workbench/api/common/extHostTypes';
 import { isUndefinedOrNull, isString } from 'vs/base/common/types';
 import { equals, coalesce } from 'vs/base/common/arrays';
 import { ILogService } from 'vs/platform/log/common/log';
@@ -23,10 +23,6 @@ import { MarkdownString } from 'vs/workbench/api/common/extHostTypeConverters';
 import { IMarkdownString } from 'vs/base/common/htmlContent';
 
 type TreeItemHandle = string;
-
-function isMarkdownString(value: any): value is vscode.MarkdownString {
-	return (value !== undefined) && value.appendCodeblock && value.appendMarkdown && value.appendText && (value.value !== undefined);
-}
 
 function toTreeItemLabel(label: any, extension: IExtensionDescription): ITreeItemLabel | undefined {
 	if (isString(label)) {
@@ -531,7 +527,7 @@ class ExtHostTreeView<T> extends Disposable {
 	}
 
 	private getTooltip(tooltip?: string | vscode.MarkdownString): string | IMarkdownString | undefined {
-		if (isMarkdownString(tooltip)) {
+		if (MarkdownStringType.isMarkdownString(tooltip)) {
 			checkProposedApiEnabled(this.extension);
 			return MarkdownString.from(tooltip);
 		}

--- a/src/vs/workbench/api/common/extHostTypes.ts
+++ b/src/vs/workbench/api/common/extHostTypes.ts
@@ -1276,7 +1276,7 @@ export class MarkdownString {
 		if (thing instanceof MarkdownString) {
 			return true;
 		}
-		return (thing !== undefined) && thing.appendCodeblock && thing.appendMarkdown && thing.appendText && (thing.value !== undefined);
+		return thing && thing.appendCodeblock && thing.appendMarkdown && thing.appendText && (thing.value !== undefined);
 	}
 }
 

--- a/src/vs/workbench/api/common/extHostTypes.ts
+++ b/src/vs/workbench/api/common/extHostTypes.ts
@@ -1271,6 +1271,13 @@ export class MarkdownString {
 		this.value += '\n```\n';
 		return this;
 	}
+
+	static isMarkdownString(thing: any): thing is vscode.MarkdownString {
+		if (thing instanceof MarkdownString) {
+			return true;
+		}
+		return (thing !== undefined) && thing.appendCodeblock && thing.appendMarkdown && thing.appendText && (thing.value !== undefined);
+	}
 }
 
 @es5ClassCompat

